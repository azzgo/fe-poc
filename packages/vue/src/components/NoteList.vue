<template>
  <div class="row between-xs">
    <div class="row shadow-1 note-card col-xs-4" v-for="note in notes" :key="note.id">
      <div class="icon" @click="onCheckNote(note)">
        <i class="material-icons">check</i>
      </div>
      <div class="col-xs-12 title">{{note.title}}</div>
      <div class="col-xs-12 value">{{note.value}}</div>
    </div>
  </div>
</template>

<style lang="less" scoped>
.note-card {
  padding: 15px;
  padding-top: 35px;
  border-radius: 2px;
  width: 100%;
  position: relative;
}

.title {
  font-size: 1.2rem;
  font-weight: bold;
  text-align: left;
  color: rgba(0, 0, 0, 0.8);
}

.value {
  text-align: left;
  font-size: 1.4rem;
  font-weight: 200;
}

.icon {
  position: absolute;
  color: black;
  border: 1px solid lightgrey;
  background-color: white;
  font-size: 30px;
  top: -10px;
  left: -10px;
  width: 40px;
  height: 40px;
  border-radius: 100%;
  cursor: pointer;
}
</style>


<script>
export default {
  props: {
    notes: Array,
  },
  methods: {
    onCheckNote(note) {
      this.$emit('onCheckNote', note)
    }
  }
}
</script>
