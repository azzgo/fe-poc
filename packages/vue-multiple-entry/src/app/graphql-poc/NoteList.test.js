import { shallowMount } from '@vue/test-utils'
import NoteList from './NoteList.vue'

test('Note List render on empty', () => {
  const comp = shallowMount(NoteList, { propsData: { notes: [] } })
  expect(comp.html()).toMatchSnapshot()
})

test('Note List render on some data', () => {
  const comp = shallowMount(NoteList, {
    propsData: {
      notes: [
        {
          id: '1',
          title: '今天吃什么',
          value: '豆花汤',
        },
        {
          id: '2',
          title: '今天看什么',
          value: '公路竞技赛',
        },
      ],
    },
  })
  expect(comp.html()).toMatchSnapshot()

  comp.find('.icon').trigger('click')
  expect(comp.emitted().onCheckNote).toBeTruthy()
  expect(comp.emitted().onCheckNote[0]).toEqual([{ id: '1', title: '今天吃什么', value: '豆花汤' }])
})
