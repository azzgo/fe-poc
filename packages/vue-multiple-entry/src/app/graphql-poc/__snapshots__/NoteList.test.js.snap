// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note List render on empty 1`] = `<div class="row between-xs"></div>`;

exports[`Note List render on some data 1`] = `
<div class="row between-xs">
  <div class="row shadow-1 note-card col-xs-4">
    <div class="icon">
      <i class="material-icons">check</i>
    </div>
    <div class="col-xs-12 title">今天吃什么</div>
    <div class="col-xs-12 value">豆花汤</div>
  </div>
  <div class="row shadow-1 note-card col-xs-4">
    <div class="icon">
      <i class="material-icons">check</i>
    </div>
    <div class="col-xs-12 title">今天看什么</div>
    <div class="col-xs-12 value">公路竞技赛</div>
  </div>
</div>
`;
